freight_forwarder/container/container.py

```diff
@@ -109,8 +109,22 @@ def commit(self, config, image_name, tag):
 
     def delete(self, remove_volumes=False, links=False, force=False):
         """
+        Delete the container on the host. Inspects the container to determine if it is in a `dead` state. This is
+        verified to take alterntaive action and not error out when deploying to a host
         """
         response = None
+
+        try:
+            if self.state()['dead']:
+                logger.warning('\'{0}\' will not be deleted. '
+                               'Container has been marked in a dead state. id: \'{1}\'. '
+                               'host: \'{2}\''.format(self.name, self.id, self.client.base_url))
+                return response
+        except APIError as e:
+            if e.response.status_code == 404:
+                logger.info('is unable to located.', extra={'formatter': 'container', 'container': self.name})
```

Comment:
I recommend you use logger.exception or similar technique to capture and log the full originating stack trace.  Throwing the trace away here means you lose potentially useful diagnostic information that would reduce debugging time.